File: server/src/main/java/com/edutech/progressive/entity/Clinic.java
package com.edutech.progressive.entity;

import java.util.ArrayList;
import java.util.List;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;

@Entity
public class Clinic {
    @Id @GeneratedValue(strategy = GenerationType.AUTO)
    private int clinicId;
    private String clinicName;
    private String location;
    private int doctorId;
    private String contactNumber;
    private int establishedYear;

    @OneToMany(mappedBy = "clinic")
    List<Doctor> doctors = new ArrayList<>();

    public Clinic() {

    }

    public Clinic(int clinicId, String clinicName, String location, int doctorId, String contactNumber,
            int establishedYear) {
        this.clinicId = clinicId;
        this.clinicName = clinicName;
        this.location = location;
        this.doctorId = doctorId;
        this.contactNumber = contactNumber;
        this.establishedYear = establishedYear;
    }

    public int getClinicId() {
        return clinicId;
    }

    public void setClinicId(int clinicId) {
        this.clinicId = clinicId;
    }

    public String getClinicName() {
        return clinicName;
    }

    public void setClinicName(String clinicName) {
        this.clinicName = clinicName;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public int getDoctorId() {
        return doctorId;
    }

    public void setDoctorId(int doctorId) {
        this.doctorId = doctorId;
    }

    public String getContactNumber() {
        return contactNumber;
    }

    public void setContactNumber(String contactNumber) {
        this.contactNumber = contactNumber;
    }

    public int getEstablishedYear() {
        return establishedYear;
    }

    public void setEstablishedYear(int establishedYear) {
        this.establishedYear = establishedYear;
    }

}

File: client/src/app/mediconnect/models/Clinic.ts
import { Doctor } from './Doctor';

export class Clinic {
    clinicId: number;
    clinicName: string;
    location: string;
    contactNumber: string;
    establishedYear: number;
    doctor: Doctor; // Many-to-One relationship (Each clinic is associated with one doctor)

    constructor(
        clinicId: number,
        clinicName: string,
        location: string,
        contactNumber: string,
        establishedYear: number,
        doctor: Doctor
    ) {
        this.clinicId = clinicId;
        this.clinicName = clinicName;
        this.location = location;
        this.contactNumber = contactNumber;
        this.establishedYear = establishedYear;
        this.doctor = doctor;
    }
}